src/omnisharp/server.ts

```diff
@@ -445,22 +448,39 @@ export class OmniSharpServer {
                 });
             }
 
-            // If there's more than one launch target, we start the server if one of the targets
-            // matches the preferred path. Otherwise, we fire the "MultipleLaunchTargets" event,
-            // which is handled in status.ts to display the launch target selector.
-            if (launchTargets.length > 1 && preferredPath) {
+            this._options = Options.Read(this.vscode);
```

Comment:
We introduced a new concept here--I'm sending you a PR to your branch that cleans this up.